pkg/controller/kubelet-config/kubelet_config_bootstrap_test.go

```diff
@@ -28,7 +28,7 @@ func TestRunKubeletBootstrap(t *testing.T) {
 				helpers.NewMachineConfigPool("custom", nil, customSelector, "v0"),
 			}
 
-			kcRaw, err := EncodeKubeletConfig(&kubeletconfigv1beta1.KubeletConfiguration{MaxPods: 100}, kubeletconfigv1beta1.SchemeGroupVersion)
+			kcRaw, err := EncodeKubeletConfig(&kubeletconfigv1beta1.KubeletConfiguration{MaxPods: 100}, kubeletconfigv1beta1.SchemeGroupVersion, runtime.ContentTypeJSON)
```

Comment:
Just curious, why are some of the tests explicitly using JSON still?

Reply:
This uses json format to set `RawExtension` filed in the test kubeletconfig object,  otherwise the test failed with the controller returning a JSON unmarshal error.
I thought it because this doc about `RawExtension`
https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/controller-revision-v1/#ControllerRevision, this field holds the unpacked data in JSON format.  So keep using json encoding for kubeletconfig only for unit/e2e testing.

Reply:
ack, makes sense, thanks!